dao/src/test/java/se/fortnox/reactivewizard/db/DaoTransactionsTest.java

```diff
@@ -60,78 +54,84 @@ public void shouldRunTwoQueriesInOneTransaction() throws SQLException {
     }
 
     @Test
-    public void shouldRunOnTransactionCompletedCallback() throws SQLException {
-        when(db.getPreparedStatement().executeBatch())
-            .thenReturn(new int[]{1, 1});
-
-        final boolean[] cbExecuted   = {false};
-        DaoObservable   daoObsWithCb = ((DaoObservable)dao.updateSuccess()).doOnTransactionCompleted(() -> cbExecuted[0] = true);
-
-        daoTransactions.executeTransaction(dao.updateSuccess(), daoObsWithCb).toBlocking().singleOrDefault(null);
-
-        assertThat(cbExecuted[0]).isTrue();
-    }
-
-    @Test
-    public void shouldRunTwoQueriesInTransactionOrderAndNotInSubscribeOrder() throws SQLException {
+    public void subscribingToDaoObservableWillResultInTwoCallsToQuery() throws SQLException {
         Observable<String> find1 = dao.find();
-        Observable<String> find2 = dao.find2();
+        Observable<String> find2 = dao.find();
 
-        daoTransactions.createTransaction(find1, find2);
+        daoTransactions.executeTransaction(find1, find2).toBlocking().subscribe();
 
-        find2.subscribe();
-        find1.toBlocking().singleOrDefault(null);
+        db.verifyConnectionsUsed(1);
+        verify(db.getConnection(), times(2)).prepareStatement(any());
 
-        Connection connection = db.getConnection();
-        InOrder    inOrder    = inOrder(connection);
+        // TODO: Is this expected? Or do we want this to be handled in some other way?
+        find2.toBlocking().singleOrDefault(null);
 
-        inOrder.verify(connection).prepareStatement("select * from test");
-        inOrder.verify(connection).prepareStatement("select * from test2");
+        db.verifyConnectionsUsed(2);
+        verify(db.getConnection(), times(1)).setAutoCommit(false);
+        verify(db.getConnection(), times(1)).commit();
+        verify(db.getConnection(), times(3)).prepareStatement("select * from test");
+        verify(db.getConnection(), timeout(500).times(2)).setAutoCommit(true);
+        verify(db.getConnection(), times(2)).close();
+        verify(db.getPreparedStatement(), times(3)).close();
+
+        // TODO: Is this expected in a transaction?
```

Comment:
Well, since transactions are returning Void then there is no point in running selects in them... But sure, it should probably execute these queries anyway. Options: Keep assert / Drop assert / Change from selects to inserts and assert some other relevant call